### DISCUSSION TITLE
What optimizer runs should we have?

### DISCUSSION THREAD

**Discussion by andreivladbrg** [sablier-labs/evm-monorepo#985] [General]
As I started working on [package tethering](https://github.com/sablier-labs/v2-core/issues/808) issue, a question came to mind, as we have different deployment configurations between `v2-core` and `v2-periphery`.

Core:

https://github.com/sablier-labs/v2-core/blob/992638415445b0e93d0bb347c8ed1fd0872c9104/foundry.toml#L18

Periphery:

https://github.com/sablier-labs/v2-periphery/blob/18d08400fa75194ac48c016e7913064eb63c32b1/foundry.toml#L20

We have two options:
1. Increase the `optimizer_runs` and use the same configuration for both deployments (we can increase it to 4000)
2. Add two foundry profiles (core/periphery) and deploy the contracts with different `optimizer_runs`

I personally prefer the first option, as I don’t find the added complexity worth it. Also, regarding option 2, if we want to use different optimizer runs for different contracts, we can use a different number of runs for NFTDescriptor (which has the biggest size), LL, LD, and LT. (I don’t find it worth it)

@sablier-labs/solidity wdyt?

**Comment by smol-ninja** [comment]
I will also vote for option (1). But why 4000? Doesn't it also exceed the limit for `NFTDescriptor`.

In that case, since higher the better, we can safely use 8,000 given LD has the highest size. And then we can use 1000 for NFTDescriptor deployment.

**Reply by andreivladbrg** [reply]
We can't increase it to 4000, I was wrong. 
We have to keep it to 1000

**Reply by PaulRBerg** [reply]
That's fine

**Comment by PaulRBerg** [comment]
Let's go with option 1 and set the highest number of optimizer runs that doesn't result into the NFT Descriptor going above the bytecode size limit.
